### PR TITLE
Convert AppDomainNative::CreateDynamicAssembly to QCall

### PR DESCRIPTION
**Pull request by clamp03**
- Get rid of the libunwind overhead by converting to QCall

### CODE REVIEW

File: src/coreclr/src/vm/appdomainnative.cpp [hunk 1]
```diff
@@ -19,33 +19,35 @@
...
+// static
+void QCALLTYPE CreateDynamicAssembly(QCall::ObjectHandleOnStack assemblyName, QCall::StackCrawlMarkHandle stackMark, INT32 access, QCall::ObjectHandleOnStack retAssembly)
 {
-    FCALL_CONTRACT;
+    QCALL_CONTRACT;
 
-    ASSEMBLYREF refRetVal = NULL;
+    BEGIN_QCALL
 
     //<TODO>
```
Comment (jkotas) [inline src/coreclr/src/vm/appdomainnative.cpp]: You are missing `GCX_COOP` here

Comment (clamp03) [reply src/coreclr/src/vm/appdomainnative.cpp]: Oops, I missed. Thank you!

File: src/coreclr/src/vm/appdomainnative.cpp [hunk 2]
```diff
@@ -19,33 +19,37 @@
 #include "clr/fs/path.h"
 using namespace clr::fs;
 
-FCIMPL3(Object*, AppDomainNative::CreateDynamicAssembly, AssemblyNameBaseObject* assemblyNameUNSAFE, StackCrawlMark* stackMark, INT32 access)
+// static
+void QCALLTYPE CreateDynamicAssembly(QCall::ObjectHandleOnStack assemblyName, QCall::StackCrawlMarkHandle stackMark, INT32 access, QCall::ObjectHandleOnStack retAssembly)
```
Comment (jkotas) [inline src/coreclr/src/vm/appdomainnative.cpp]: You are missing `AppDomainNative::` that is making all builds to fail.

Comment (clamp03) [reply src/coreclr/src/vm/appdomainnative.cpp]: Sorry. I tested with old dotnet/coreclr, but there are some mistakes when I apply it to dotnet/runtime by manually.

File: src/coreclr/src/vm/ecalllist.h [hunk 3]
```diff
@@ -536,7 +536,7 @@ FCFuncStart(gAssemblyFuncs)
 FCFuncEnd()
 
 FCFuncStart(gAssemblyBuilderFuncs)
-    FCFuncElement("nCreateDynamicAssembly", AppDomainNative::CreateDynamicAssembly)
+    QCFuncElement("CreateDynmicAssembly", AppDomainNative::CreateDynamicAssembly)
```
Comment (jkotas) [inline src/coreclr/src/vm/ecalllist.h]:
```suggestion
    QCFuncElement("CreateDynamicAssembly", AppDomainNative::CreateDynamicAssembly)
```

Comment (clamp03) [reply src/coreclr/src/vm/ecalllist.h]: Thank you.

**Review by jkotas — APPROVED** [review]

**Pull request merged**